podpac/core/data/datasource.py

```diff
@@ -295,23 +339,23 @@ def eval(self, coordinates, output=None):
                     extra.append(c.name)
         coordinates = coordinates.drop(extra)
 
-        # set input coordinates to evaluated coordinates
-        # TODO move this if WCS can be updated to support
-        self._evaluated_coordinates = coordinates
-        return self._eval(coordinates, output=output)
+        # store input coordinates to evaluated coordinates
+        self._evaluated_coordinates = deepcopy(coordinates)
+
+        # transform coordinates into native crs if different
+        if self.native_coordinates.crs != coordinates.crs:
+            coordinates = coordinates.transform(self.native_coordinates.crs)
 
-    @node_eval
-    def _eval(self, coordinates, output=None):
         # intersect the native coordinates with requested coordinates
         # to get native coordinates within requested coordinates bounds
         # TODO: support coordinate_index_type parameter to define other index types
         self._requested_source_coordinates, self._requested_source_coordinates_index = \
             self.native_coordinates.intersect(coordinates, outer=True, return_indices=True)
 
-        # If requested coordinates and native coordinates do not intersect, shortcut with nan UnitsDataArary
+        # if requested coordinates and native coordinates do not intersect, shortcut with nan UnitsDataArary
         if np.prod(self._requested_source_coordinates.shape) == 0:
```

Comment:
For future reference, there should be a `size` attribute on coordinates (which just does this np.prod for you).